src/pb_stub.cc

```diff
@@ -1116,9 +1116,6 @@ Stub::ParentToStubMQMonitor()
           *(ipc_message->ResponseMutex())};
       response_batch->waiting_on_stub = true;
       ipc_message->ResponseCondition()->notify_all();
-      while (response_batch->waiting_on_stub) {
-        ipc_message->ResponseCondition()->wait(lock);
-      }
```

Comment:
I think we need this call since the response batch could be deallocated from shared memory. Feel free to correct me if I'm wrong here.

Reply:
I was wondering if you mean that as soon as the stub process reaches this line: https://github.com/triton-inference-server/python_backend/blob/main/src/pb_stub.cc#L1118, the parent process will exit from the function and hence the response batch may be deallocated?
If that's the case, I think since the stub process already retrieves the content from the response batch, forms an infer response and [store it into the response_iterator](https://github.com/triton-inference-server/python_backend/blob/main/src/pb_stub.cc#L1099-L1112) before calling `notify_all()` to inform the parent process, the response batch will not go out of scope before we finish processing it in the stub process. Please let me know if I'm misunderstanding something from here.